converter/src/main/java/org/dswarm/converter/morph/MorphScriptBuilder.java

```diff
@@ -344,66 +348,56 @@ private void createLookupTable(final Element maps, final Component transformatio
 				if (LOOKUP_FUNCTIONS.contains(component.getFunction().getName()) && componentParameterMapping != null) {
 
 					final Element map = doc.createElement(METAMORPH_ELEMENT_SINGLE_MAP);
+					map.setAttribute("name", component.getName());
 					maps.appendChild(map);
 
 					for (final Entry<String, String> parameterMapping : componentParameterMapping.entrySet()) {
 
-						if (parameterMapping.getKey() != null && parameterMapping.getValue() != null) {
-
-							switch (parameterMapping.getKey()) {
-
-								case "in":
```

Comment:
"in" and "map" could also be a constant here, or? (i.e. private static final ...)